bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperSTTService.java

```diff
@@ -553,7 +551,120 @@ private void backgroundRecognize(WhisperJNI whisper, WhisperContext ctx, Whisper
+        try {
+            whisper = getWhisper();
+            ctx = getContext();
+            logger.debug("Creating whisper state...");
+            state = whisper.initState(ctx);
+            logger.debug("Whisper state created");
+            WhisperFullParams params = getWhisperFullParams(ctx, language);
+
+            // convert to local whisper format (float)
+            float[] floatArray = new float[audioSamples.length];
+            for (int i = 0; i < audioSamples.length; i++) {
+                floatArray[i] = Float.min(1f, Float.max((float) audioSamples[i] / ((float) Short.MAX_VALUE), -1f));
+            }
+
+            long execStartTime = System.currentTimeMillis();
+            var result = whisper.fullWithState(ctx, state, params, floatArray, audioSamplesOffset);
+            logger.debug("whisper ended in {}ms with result code {}", System.currentTimeMillis() - execStartTime,
+                    result);
+            // process result
+            if (result != 0) {
+                throw new STTException("Cannot use whisper locally, result code: " + result);
+            }
+            int nSegments = whisper.fullNSegmentsFromState(state);
+            logger.debug("Available transcription segments {}", nSegments);
+            if (nSegments == 1) {
+                return whisper.fullGetSegmentTextFromState(state, 0);
+            } else if (nSegments == 0 && config.singleUtteranceMode) {
+                logger.debug("Single utterance mode and no results, ending transcription");
+                return null;
+            } else {
+                // non reachable
+                logger.warn("Whisper should be configured in single segment mode {}", nSegments);
+                return null;
+            }
+        } catch (IOException e) {
+            if (state != null) {
+                state.close();
+            }
+            throw new STTException("Cannot use whisper locally", e);
+        } finally {
+            if (releaseContext && ctx != null) {
+                ctx.close();
+            }
+        }
+    }
+
+    private String recognizeAPI(int audioSamplesOffset, short[] audioStream, String language) throws STTException {
+
+        // convert to byte array, Each short has 2 bytes
+        int size = audioSamplesOffset * 2;
+        ByteBuffer byteArrayBuffer = ByteBuffer.allocate(size).order(ByteOrder.LITTLE_ENDIAN);
+        for (int i = 0; i < audioSamplesOffset; i++) {
+            byteArrayBuffer.putShort(audioStream[i]);
+        }
+        javax.sound.sampled.AudioFormat jAudioFormat = new javax.sound.sampled.AudioFormat(
+                javax.sound.sampled.AudioFormat.Encoding.PCM_SIGNED, WHISPER_SAMPLE_RATE, 16, 1, 2, WHISPER_SAMPLE_RATE,
+                false);
+        byte[] byteArray = byteArrayBuffer.array();
+
+        try {
+            AudioInputStream audioInputStream = new AudioInputStream(new ByteArrayInputStream(byteArray), jAudioFormat,
+                    size);
```

Comment:
I found the problem that causes the `--convert` option to be required, this should be the size of the stream in samples (not in bytes), so passing `audioSamplesOffset` instead of `size` writes correctly the header and fixes the problem.
```suggestion
                    audioSamplesOffset);
```

Reply:
Great ! Thank you and sorry for letting it slip past my tests.